rs/replicated_state/src/canister_state/execution_state.rs

```diff
@@ -477,6 +478,14 @@ pub struct ExecutionState {
 
     /// Round-robin across canister method types.
     pub next_scheduled_method: NextScheduledMethod,
+
+    /// Checks if execution is in Wasm64 mode. This field is used as a cache to avoid
+    /// parsing the wasm module to determine if it is a Wasm64 module.
+    /// Should not be accessed directly, use `is_wasm64()` method instead.
+    /// Also this field does not need to be saved in the checkpoint/state
+    /// as it can always be recomputed from the wasm module.
+    #[validate_eq(Ignore)]
+    pub is_wasm64: Option<bool>,
```

Comment:
Given the comment above, this shouldn't be pub, and maybe be called `is_wasm64_cache` for clarity.

Reply:
it's now saved in the state. please have another look.